catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPostProcessPlugin.java

```diff
@@ -0,0 +1,363 @@
+        }
+
+        addDerivedResourceAttribute(newProcessResourceItem);
+        return newProcessResourceItem;
+    }
+
+    private static boolean isNitfMimeType(String rawMimeType) {
+        try {
+            return MetacardFactory.MIME_TYPE.match(rawMimeType);
+        } catch (MimeTypeParseException e) {
+            LOGGER.debug("unable to compare mime types: {} vs {}",
+                    MetacardFactory.MIME_TYPE,
+                    rawMimeType);
+        }
+
+        return false;
+    }
+
+    private static BufferedImage renderImage(ProcessResource processResource)
+            throws IOException, ParseException, NitfFormatException {
+
+        final ThreadLocal<BufferedImage> bufferedImage = new ThreadLocal<>();
+
+        if (processResource != null && processResource.getInputStream() != null) {
+            NitfRenderer renderer = new NitfRenderer();
+
+            new NitfParserInputFlow().inputStream(processResource.getInputStream())
+                    .allData()
+                    .forEachImageSegment(segment -> {
+                        if (bufferedImage.get() == null) {
+                            try {
+                                bufferedImage.set(renderer.render(segment));
+                            } catch (IOException e) {
+                                LOGGER.debug(e.getMessage(), e);
+                            }
+                        }
+                    })
+                    .end();
+        }
+
+        return bufferedImage.get();
+    }
+
+    private static void addThumbnailToMetacard(Metacard metacard, BufferedImage bufferedImage) {
+        try {
+            byte[] thumbnailImage = scaleImage(bufferedImage, THUMBNAIL_WIDTH, THUMBNAIL_HEIGHT);
+
+            if (thumbnailImage.length > 0) {
+                metacard.setAttribute(new AttributeImpl(Core.THUMBNAIL, thumbnailImage));
+            }
+        } catch (IOException e) {
+            LOGGER.debug(e.getMessage(), e);
+        }
+    }
+
+    private static ProcessResource createDerivedImage(String id, String qualifier,
+            BufferedImage image, Metacard metacard, int maxWidth, int maxHeight)
+            throws IOException {
+        byte[] overviewBytes = scaleImage(image, maxWidth, maxHeight);
+        ByteSource source = ByteSource.wrap(overviewBytes);
+
+        return new ProcessResourceImpl(id,
+                source.openStream(),
+                IMAGE_JPEG,
+                buildDerivedImageTitle(metacard.getTitle(), qualifier),
+                overviewBytes.length,
+                qualifier);
+    }
+
+    private static String buildDerivedImageTitle(String title, String qualifier) {
+        String rootFileName = FilenameUtils.getBaseName(title);
+
+        // title must contain some alphanumeric, human readable characters, or use default filename
+        if (StringUtils.isNotBlank(rootFileName) && StringUtils.isNotBlank(rootFileName.replaceAll(
+                "[^A-Za-z0-9]",
```

Comment:
`Pattern.compile` could be used here too.  Might be good as a global constant as well